scripts/generateAssetData/generateRelatedAssetIndex/generateRelatedAssetIndex.ts

```diff
@@ -0,0 +1,213 @@
+import type { AssetId } from '@shapeshiftoss/caip'
+import { FEE_ASSET_IDS, fromAssetId } from '@shapeshiftoss/caip'
+import { isEvmChainId } from '@shapeshiftoss/chain-adapters'
+import type { AssetsById } from '@shapeshiftoss/types'
+import assert from 'assert'
+import axios from 'axios'
+import axiosRetry from 'axios-retry'
+import { Presets, SingleBar } from 'cli-progress'
+import fs from 'fs'
+import { isNull } from 'lodash'
+import isUndefined from 'lodash/isUndefined'
+import path from 'path'
+
+import { zerionImplementationToMaybeAssetId } from './mapping'
+import { zerionFungiblesSchema } from './validators/fungible'
+
+const ZERION_BASE_URL = 'https://api.zerion.io/v1'
+const BATCH_SIZE = 100
+
+const axiosInstance = axios.create()
+axiosRetry(axiosInstance, { retries: 5, retryDelay: axiosRetry.exponentialDelay })
+
+const isSome = <T>(option: T | null | undefined): option is T =>
+  !isUndefined(option) && !isNull(option)
+
+const chunkArray = <T>(array: T[], chunkSize: number) => {
+  const result = []
+  for (let i = 0; i < array.length; i += chunkSize) {
+    const chunk = array.slice(i, i + chunkSize)
+    result.push(chunk)
+  }
+  return result
+}
+
+const createThrottle = ({
+  capacity,
+  costPerReq,
+  drainPerInterval,
+  intervalMs,
+}: {
+  capacity: number
+  costPerReq: number
+  drainPerInterval: number
+  intervalMs: number
+}) => {
+  let currentLevel = 0
+
+  setInterval(() => {
+    currentLevel = Math.max(0, currentLevel - drainPerInterval)
+  }, intervalMs)
+
+  const throttle = async () => {
+    let isFull = currentLevel + costPerReq >= capacity
+    while (isFull) {
+      await new Promise(resolve => setTimeout(resolve, intervalMs))
+      isFull = currentLevel + costPerReq >= capacity
+    }
+  }
+
+  return throttle
+}
+
+const getRelatedAssetIds = async (
+  assetId: AssetId,
+  assetData: AssetsById,
+): Promise<{ relatedAssetIds: AssetId[]; relatedAssetKey: AssetId } | undefined> => {
+  assert(process.env.ZERION_API_KEY !== undefined, 'Missing Zerion API key')
```

Comment:
This is missing in env @woodenfurniture - will either need to be added in .env if we have a rationale for fetching directly from Zerion, but given how this is throttled, it looks like this is unintentional and we should fetch from `https://zerion.shapeshift.com` ?

Reply:
absolutely correct, we must no fetch from our throttled url for this - ill add a comment.

Also, RE the .env variable, we definitely wont be committing the api key to source control, I'll need to update the readme on this point. The idea is we request the api key from WS lead and run the script